src/metabase/integrations/slack.clj

```diff
@@ -403,17 +432,27 @@
       (log/debug "Uploaded image" <>))))
 
 (mu/defn post-chat-message!
-  "Calls Slack API `chat.postMessage` endpoint and posts a message to a channel. `attachments` should be serialized
-  JSON."
+  "Calls Slack API `chat.postMessage` endpoint and posts a message to a channel.
+   Can accept either `attachments` (as serialized JSON) or `blocks` (as an array)."
   [channel-id  :- ms/NonBlankString
    text-or-nil :- [:maybe :string]
-   & [attachments]]
+   & [attachments-or-blocks]]
```

Comment:
I'll leave it up to you, but what do you think about updating function signature to accept a map in the 3rd arg position, with optional `:blocks` and `:attachments` keys? Overloading the 3rd arg position to be attachment data or blocks seems to complect the two separate concepts.

That would improve the readability when computing params in the `cond->` on 441.